### ISSUE TITLE
deps.json file has empty compilationOptions section

### ISSUE THREAD

**Issue by davidfowl** [dotnet/sdk#3251]
Likely a result of the migration to the new JSON APIs

```JSON
{
  "runtimeTarget": {
    "name": ".NETCoreApp,Version=v3.0",
    "signature": ""
  },
  "compilationOptions": {},
  "targets": {
    ".NETCoreApp,Version=v3.0": {
      "ConsoleApp1/1.0.0": {
        "runtime": {
          "ConsoleApp1.dll": {}
        }
      }
    }
  },
  "libraries": {
    "ConsoleApp1/1.0.0": {
      "type": "project",
      "serviceable": false,
      "sha512": ""
    }
  }
}
```

cc @ahsonkhan @nguerrera

**Comment by nguerrera** [comment]
Cc @dsplaisted

**Comment by dsplaisted** [comment]
Does this have any impact?  Are there supposed to be compilation options recorded in this case?

**Comment by davidfowl** [comment]
No functionality impact but it should be cleaned up as it's unnecessary and frankly strange to see.

**Comment by ahsonkhan** [comment]
> Likely a result of the migration to the new JSON APIs

I don't think the migration affected this. At least, looking at the tests, this was by-design before too:
https://github.com/dotnet/sdk/blob/b73819e683af64bd19c961973c77b2239cd20e5d/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs#L63

https://github.com/dotnet/sdk/blob/8cb31b0fd0bb5d36b9781442865f2e26cf6fbedc/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/simple.dependencies.deps.json#L6

**Comment by davidfowl** [comment]
Make a new console app with .net core 2.1 or 2.2, does it have the compilationOptions?

**Comment by nguerrera** [comment]
Just tried it with an old 2.1.4xx sdk and a 2.1 app. It does the same thing. Seems it's always been this way.

**Comment by davidfowl** [comment]
Omg then we should fix it...

**Comment by nguerrera** [comment]
If we're going after brevity, I'd like to see the empty signature node removed. That isn't used by anything, it's an artifact of some servicing thing before my time that never saw the light of day. Also, the empty sha512...

**Comment by dsplaisted** [comment]
We use the Microsoft.Extensions.DependencyModel APIs to write the deps file, so it's likely that we'd need a change in that library in order to make these changes.

**Comment by livarcocc** [comment]
That's right. This is coming from the DependencyModel. Is the plan here to remove CompilationOptions from the DependencyModel entirey because this thing is never used or just to not create the empty node?

**Comment by dsplaisted** [comment]
AFAIK there are still some scenarios where CompilationOptions is used, so the proposal would be to not include it in the deps file if it's empty.
